File: Scientist4JCore/src/test/java/com/github/rawls238/scientist4k/TestPublishExperiment.java
package com.github.rawls238.scientist4k;

import com.github.rawls238.scientist4k.metrics.MetricsProvider;

import static org.assertj.core.api.Assertions.assertThat;

public class TestPublishExperiment<Integer> extends Experiment<Integer> {
  TestPublishExperiment(String name, MetricsProvider<?> metricsProvider) {
    super(name, metricsProvider);
  }

  @Override
  protected void publish(Result<Integer> r) {
    assertThat(r.getCandidate().get().getDuration()).isGreaterThan(0L);
    assertThat(r.getControl().getDuration()).isGreaterThan(0L);
  }

}

File: Scientist4KCore/src/test/kotlin/com/github/squirrelgrip/scientist4k/TestPublishExperiment.kt
package com.github.squirrelgrip.scientist4k

import com.github.squirrelgrip.scientist4k.metrics.MetricsProvider
import com.github.squirrelgrip.scientist4k.model.Result
import org.assertj.core.api.Assertions

class TestPublishExperiment<Integer>(name: String, metricsProvider: MetricsProvider<*>) : Experiment<Integer>(name, metricsProvider) {
    override fun publish(result: Result<Integer>) {
        Assertions.assertThat(result.candidate?.duration).isGreaterThan(0L)
        Assertions.assertThat(result.control.duration).isGreaterThan(0L)
    }
}

class TestPublishControlledExperiment<Integer>(name: String, metricsProvider: MetricsProvider<*>) : ControlledExperiment<Integer>(name, metricsProvider) {
    override fun publish(result: Result<Integer>) {
        Assertions.assertThat(result.candidate?.duration).isGreaterThan(0L)
        Assertions.assertThat(result.control.duration).isGreaterThan(0L)
    }
}